File: src/main/java/com/freeplayer/model/Genero.java
package com.freeplayer.model;

public class Genero {
    private int idGenero;
    private String nombreGenero;

    public Genero() {
    }
    public Genero(int idGenero, String nombreGenero) {
        this.idGenero = idGenero;
        this.nombreGenero = nombreGenero;
    }

    public int getIdGenero() {
        return idGenero;
    }

    public void setIdGenero(int idGenero) {
        this.idGenero = idGenero;
    }

    public String getNombreGenero() {
        return nombreGenero;
    }

    public void setNombreGenero(String nombreGenero) {
        this.nombreGenero = nombreGenero;
    }

    @Override
    public String toString() {
        return "Genero{" +
                "idGenero=" + idGenero +
                ", nombreGenero='" + nombreGenero + '\'' +
                '}';
    }
}

File: app/src/main/java/com/pmk/freeplayer/domain/model/audio/Genero.kt
package com.pmk.freeplayer.domain.model.audio

import java.util.Locale

@JvmInline
value class Genero(val valor: String) {
	
	val normalizado: String
		get() = valor.trim()
			.lowercase()
			.replaceFirstChar { if (it.isLowerCase()) it.titlecase(Locale.ROOT) else it.toString() }
	
	companion object {
		val DESCONOCIDO = Genero("Desconocido")
		val OTRO = Genero("Otro")
	}
}